### ISSUE TITLE
Wrong "Unnecessary assignment of a value"

### ISSUE THREAD

**Issue by fbrosseau** [dotnet/roslyn#40634]
Hello

**Version Used**: 
16.4.2

**Steps to Reproduce**:

The following code on 16.4.2 in a brand new empty netcore project triggers an incorrect diagnostic on line `callWriteLine = false;`, saying that assignment is unnecessary, and the suggested Quick Action actually deletes the line, which is wrong. My original code had a real condition inside the 'if', but apparently even an always-false 'if' triggers this bug.

Sample
```
class Program
{
    public static void Test()
    {
        bool callWriteLine = true;
        try
        {
            if (false)
                throw null;
            callWriteLine = false;
        }
        finally
        {
            if (callWriteLine)
                System.Console.WriteLine("!");
        }
    }
}
```

Interestingly enough, the throw is unrelated. Even the following sample has the same issue (even though it even successfully identifies `Console.WriteLine("hello")` as unreachable!)

```
    public static void Test()
    {
        bool callWriteLine = true;
        try
        {
            if (false)
                Console.WriteLine("hello");
            callWriteLine = false;
        }
        finally
        {
            if (callWriteLine)
                Console.WriteLine("!");
        }
    }
```

**Comment by fbrosseau** [comment]
Interestingly enough, the throw is unrelated. Even the following sample has the same issue (and it even does identify `Console.WriteLine("hello")` as unreachable!)

```
    public static void Test()
    {
        bool callWriteLine = true;
        try
        {
            if (false)
                Console.WriteLine("hello");
            callWriteLine = false;
        }
        finally
        {
            if (callWriteLine)
                Console.WriteLine("!");
        }
    }
```

**Comment by huoyaoyuan** [comment]
Looks like #39755 . You may see this fixed when 16.5 Preview 2 releases.

**Comment by fbrosseau** [comment]
Yes that's probably it. I don't have P2 however : )

**Comment by jinujoseph** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/39755
